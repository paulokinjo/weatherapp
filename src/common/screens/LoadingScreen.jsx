import CircularProgress from '@material-ui/core/CircularProgress';
import React from 'react';
import loadingScreenStyles from './styles/loadingScreen.styles';

const LoadingScreen = () => {
  const classes = loadingScreenStyles();
  return (
    <div className={classes.spinner}>
      <CircularProgress />
      <p>Loading...</p>
    </div>
  );
};

export default LoadingScreen;
