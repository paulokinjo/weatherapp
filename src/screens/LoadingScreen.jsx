import CircularProgress from '@material-ui/core/CircularProgress';
import React from 'react';

const LoadingScreen = () => {
  return (
    <div
      style={{
        position: 'fixed',
        top: '45%',
        left: '45%',
      }}
    >
      <CircularProgress />
      <span style={{ paddingLeft: '20px' }}>Loading...</span>
    </div>
  );
};

export default LoadingScreen;
