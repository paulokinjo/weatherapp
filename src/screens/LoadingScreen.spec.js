import LoadingScreen from './LoadingScreen';
import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

describe('LoadingScreen', () => {
  const loadingScreen = shallow(<LoadingScreen />);

  describe('Layout', () => {
    it('should render properly', () => {
      expect(loadingScreen).toMatchSnapshot();
    });
  });

  describe('Lifecycle', () => {
    it('should show loading message', () => {
      const { queryByText } = render(<LoadingScreen />);

      const loadingMessage = queryByText('Loading...');

      expect(loadingMessage).toBeInTheDocument();
    });

    it('should show loading icon', () => {
      const { container } = render(<LoadingScreen />);

      const circle = container.querySelector('circle');

      expect(circle).toBeInTheDocument();
    });
  });
});
