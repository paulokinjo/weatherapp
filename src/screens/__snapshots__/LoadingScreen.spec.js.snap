// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen Layout should render properly 1`] = `ShallowWrapper {}`;
